sources/web/datalab/settings.ts

```diff
@@ -118,8 +118,54 @@ function copyDefaultUserSettings(userId: string) {
   const defaultUserSettingsPath = path.join(__dirname, 'config', DEFAULT_USER_SETTINGS_FILE);
   console.log('Copying default settings: ' + defaultUserSettingsPath + ' to: ' + userSettingsPath);
   // Copy the default user settings file into user's directory.
-  const data = fs.readFileSync(defaultUserSettingsPath);
-  fs.writeFileSync(userSettingsPath, data);
+  const defaultUserSettings = fs.readFileSync(defaultUserSettingsPath, {encoding: 'utf8'});
+  const initialUserSettings = process.env.DATALAB_INITIAL_USER_SETTINGS;
+  const mergedUserSettings : string = initialUserSettings ?
+      mergeUserSettings(defaultUserSettings, initialUserSettings) : defaultUserSettings;
+  fs.writeFileSync(userSettingsPath, mergedUserSettings);
+  // writeFileSync does not return a status; let's see if it wrote a file.
+  if (!fs.existsSync(userSettingsPath)) {
+    console.log('Failed to write new user settings file ' + userSettingsPath);
```

Comment:
Ah, you're right. I guess I'm fine with the console log calls then.
I don't see a lot of value in splitting the file, but feel free to do it if you think it cleans it up.